src/Http/Controllers/Auth/SocialiteController.php

```diff
@@ -0,0 +1,57 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Blumilk\Meetup\Core\Http\Controllers\Auth;
+
+use Blumilk\Meetup\Core\Http\Controllers\Controller;
+use Blumilk\Meetup\Core\Models\User;
+use Laravel\Socialite\Facades\Socialite;
+
+class SocialiteController extends Controller
+{
+    public function redirectToGoogle()
+    {
+        return Socialite::driver("google")->redirect();
+    }
+    public function handleGoogleCallback()
+    {
+        $user = Socialite::driver("google")->user();
+
+        $this->registerOrLogin($user);
+
+        return redirect(route("home"));
+    }
+    public function redirectToFacebook()
+    {
+        return Socialite::driver("facebook")->redirect();
+    }
+    public function handleFacebookCallback()
+    {
+        $user = Socialite::driver("google")->user();
+
+        $this->registerOrLogin($user);
+
+        return redirect(route("home"));
+    }
+
+    protected function registerOrLogin($data)
+    {
+        $user = User::where("email", $data["email"])->first();
+        if (!$user) {
+            $user = User::create([
+                "name" => $data["name"],
+                "email" => $data["email"],
+                "provider_id" => $data["id"],
+            ]);
+        }
+
+        $token = $user->createToken("AccessToken")->plainTextToken;
+        $response = [
+            "user" => $user["email"],
+            "auth_token" => $token,
```

Comment:
Are we creating SPA or monolith app?

Reply:
I think we should create SPA app

Reply:
I see in another [PR](https://github.com/blumilksoftware/meetup/pull/29) app with normal blades. Had you and your team decision in that topic? @MrHDOLEK @MichalMyskow ?

Reply:
I spoke to the team and decided that we were creating a monolith.